File: Selenium/Activities/Activity8.py
# Import required packages from selenium
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver using WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:

    # Set Explicit wait of 3 Secs
    wait = WebDriverWait(driver, 10)

    # Launch the URL and get the title
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")
    home_title=driver.title
    print ("Title of the Home Page is : {}".format(home_title))

    # Check if the Dynamic Checkbox is displayed
    check_box = driver.find_element(By.ID, "dynamicCheckbox")
    print("Dynamic Checkbox Exists : {}".format(check_box.is_displayed()))

    # Click the toggle Button
    toggle_button = driver.find_element(By.ID, "toggleCheckbox")
    toggle_button.click()

    # Wait till the checkbox disappears
    wait.until(expected_conditions.invisibility_of_element(check_box))

    # Check if the Dynamic Checkbox is disappeared
    print("Dynamic Checkbox Exists : {}".format(check_box.is_displayed()))

    # Click the toggle Button Again
    toggle_button.click()

    # Check if the Dynamic Checkbox is disappeared
    print("Dynamic Checkbox Exists : {}".format(check_box.is_displayed()))

    # Wait for 5 Seconds and Close the Browser
    time.sleep(3)

    # Close the Browser
    driver.close()

File: Selenium/Activities/Activity8.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity8 {
    public static void main(String[] args) throws InterruptedException {
        //Setup GekoDriver
        WebDriverManager.firefoxdriver().setup();

        //Create driver Object for Firefox Browser
        WebDriver driver = new FirefoxDriver();

        //Set Explicit wait of 5 Secs
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofMillis(5000));

        //Open the url
        driver.get("https://v1.training-support.net/selenium/dynamic-controls");

        //Get the Title
        String title = driver.getTitle();
        System.out.print("Title of the Page is : " + title + "\n");

        // Check if the Dynamic Checkbox is displayed
        WebElement checkBox = driver.findElement(By.id("dynamicCheckbox"));
        System.out.println("Dynamic Checkbox Exists : " + checkBox.isDisplayed());

        // Click the toggle Button
        WebElement toggleButton = driver.findElement(By.id("toggleCheckbox"));
        toggleButton.click();

        // Wait till the checkbox disappears
        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.id("dynamicCheckbox")));

        // Check if the Dynamic Checkbox is disappeared
        System.out.println("Dynamic Checkbox Exists : " + checkBox.isDisplayed());

        // Click the toggle Button Again
        toggleButton.click();

        // Check if the Dynamic Checkbox is disappeared
        System.out.println("Dynamic Checkbox Exists : " + checkBox.isDisplayed());

        //Wait for 5 Seconds and Close the Browser
        Thread.sleep(5000);
        driver.close();
    }
}
